### ISSUE TITLE
iOS crash `0xda20 WrappedMediaPlayer.setSourceUrl` in `WrappedMediaPlayer.swift:2124`

### ISSUE THREAD

**Issue by RomanSoviak** [bluefireteam/audioplayers#1338]
Sometimes I get error from iOS
` 0xda20 WrappedMediaPlayer.setSourceUrl(url:isLocal:completer:) + 2124 (WrappedMediaPlayer.swift:2124)`
<img width="1089" alt="Screenshot 2022-11-29 at 12 57 24" src="https://user-images.githubusercontent.com/26696190/204511622-1d1a10bc-8ecf-44be-89fa-9981f78cc003.png">

```
var _audioPlayer = AudioPlayer();
       await _audioPlayer.play(UrlSource(url));
       await _audioPlayer.resume();
```

**Comment by yashpalzala** [comment]
@RomanSoviak the logs suggests that the crash is happening after setting the source Url. 
I would highly recommend you to once check if the source  url is not empty and does not contain any space or special characters.

**Comment by RomanSoviak** [comment]
@yashpalzala no, I am absolutely confident that URL is OK, I tried just_audio lib and it works without correctly, IDK what's wrong with that.  I can close this issue if this is not reproducible

**Comment by bhaveshjio** [comment]
@RomanSoviak  facing same issue on release mode iOS build. Have you found any work around?

**Comment by RomanSoviak** [comment]
@bhaveshjio I replaced this library with [just_audio](https://pub.dev/packages/just_audio) lib. Just_audio works more slowly but without crashing

**Comment by Gustl22** [comment]
@RomanSoviak we need a full issue description inclusive sources and player info etc. We cannot help otherwise as we cannot guess the inputs. Can you provide the url or upload the sample file? And fill the issue template correctly? That would help us a lot :)

**Comment by Fudal** [comment]
I have the same problem on iOS device.

**Comment by Gustl22** [comment]
@Fudal can you provide all the details in the [issue template](https://github.com/bluefireteam/audioplayers/issues/new?assignees=&labels=bug%2Ctriage&template=bug-report.yml), to help fixing this problem? 
Or just open a new issue, otherwise we cannot help or fix it.

**Comment by RomanSoviak** [comment]
I cannot send any info due to NDA, I'm just using different multiple audios from Firebase storage and after multiple playbacks of them(~100) it stopped working and in logs I see this error

**Comment by Gustl22** [comment]
Most likely fixed via #1625. An error occured if source was not parsable.
